src/App.tsx

```diff
@@ -1,10 +1,54 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import { getTodos } from './api/todos';
+import { TodosError } from './components/TodosError/TodosError';
+import { TodosFooter } from './components/TodosFooter/TodosFooter';
+import { TodosHeader } from './components/TodosHeader/TodosHeader';
+import { TodosList } from './components/TodosList/TodosList';
+import { Filter } from './types/Filter';
+import { Todo } from './types/Todo';
 import { UserWarning } from './UserWarning';
 
-const USER_ID = 0;
+const USER_ID = 11331;
+
+const ERROR_ADD = 'Unable to add a todo';
+// const ERROR_DELETE = 'Unable to delete a todo';
+// const ERROR_UPDATE = 'Unable to update a todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState(Filter.All);
+  const [error] = useState(ERROR_ADD);
+
+  function getVisibleTodos(filterType: Filter): Todo[] {
+    switch (filterType) {
+      case Filter.Active:
+        return todos.filter(todo => !todo.completed);
+      case Filter.Comleted:
+        return todos.filter(todo => todo.completed);
+      default:
+        return todos;
+    }
+  }
```

Comment:
no need to reassign this function on each render. Move it to the helper file and import it here